src/App.tsx

```diff
@@ -1,26 +1,164 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, useRef } from 'react';
 import { UserWarning } from './UserWarning';
+import { getTodos, USER_ID, postTodo, deleteTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './Components/TodoList/TodoList';
+import { Footer } from './Components/Footer/Footer';
+import { Header } from './Components/Header/Header';
+import cn from 'classnames';
 
-const USER_ID = 0;
+export enum SelectOption {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [option, setOption] = useState(SelectOption.All);
+  const [loading, setLoading] = useState(false);
+  const [titleTodo, setTitleTodo] = useState('');
+  const [inputTodo, setInputTodo] = useState(true);
+  const inputFocus = useRef<HTMLInputElement>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'))
+      .finally(() => {
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (inputFocus.current && inputTodo) {
+      inputFocus.current.focus();
+    }
+  }, [inputTodo]);
+
+  const filteredTodos = todos.filter(todo => {
+    switch (option) {
+      case SelectOption.Active:
+        return !todo.completed;
+      case SelectOption.Completed:
+        return todo.completed;
+      case SelectOption.All:
+      default:
+        return true;
+    }
+  });
+
+  const todosLength = todos.filter(todo => !todo.completed).length;
+
+  const checkCompletedTodos = todos.some(todo => todo.completed === true);
+
+  function addTodo({ title, userId, completed }: Todo) {
```

Comment:
It's important to stick to the same way of creating functions everywhere, fix it everywhere
```suggestion
  const addTodo = ({ title, userId, completed }: Todo) => {
```